src/features/comments.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../types/Comment';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const initComments = createAsyncThunk(
+  'comments/fetch',
+  (postId: number) => {
+    return getPostComments(postId);
+  },
+);
+
+export const addComment = createAsyncThunk(
+  'comments/addComment',
+  async (comment: Omit<Comment, 'id'>) => {
+    const value = await createComment(comment);
+
+    return value;
+  },
+);
+
+export const removeComment = createAsyncThunk(
+  'comments/deleteComment',
+  async (commentId: number) => {
+    await deleteComment(commentId);
+
+    return commentId;
+  },
+);
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(initComments.pending, state => {
+      state.loaded = false;
+      state.hasError = false;
+    });
+
+    builder.addCase(
+      initComments.fulfilled,
+      (state, action: PayloadAction<Comment[]>) => {
+        state.comments = action.payload;
+        state.loaded = true;
+      },
+    );
+
+    builder.addCase(initComments.rejected, state => {
+      state.loaded = true;
+      state.hasError = true;
+    });
+
+    builder.addCase(
+      addComment.fulfilled,
+      (state, action: PayloadAction<Comment>) => {
+        state.comments.push(action.payload);
+      },
+    );
+
+    builder.addCase(addComment.rejected, state => {
+      state.hasError = true;
+    });
+
+    builder
+      .addCase(removeComment.fulfilled, (state, action) => {
+        state.comments = state.comments.filter(
+          comment => comment.id !== action.payload,
+        );
+      })
+      .addCase(removeComment.rejected, state => {
+        state.hasError = true;
+      });
```

Comment:
In the `removeComment.rejected` case, similar to `addComment.rejected`, consider implementing a more detailed error handling strategy to inform the user about what went wrong.